File: cse_app/lib/models/service.dart
class Service {
  final String id;
  final String name;
  final String? description;
  final double? price;
  final bool isActive;
  final List<String> taskIds;
  final String createdAt;
  final String updatedAt;

  Service({
    required this.id,
    required this.name,
    this.description,
    this.price,
    required this.isActive,
    this.taskIds = const [],
    required this.createdAt,
    required this.updatedAt,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'],
      price: (json['price'] as num?)?.toDouble(),
      isActive: json['is_active'] ?? true,
      taskIds: (json['task_ids'] as List?)?.cast<String>() ?? [],
      createdAt: json['created_at'] ?? '',
      updatedAt: json['updated_at'] ?? '',
    );
  }
}

File: backend/app/models/service.py
from datetime import datetime, timezone
from typing import Optional
from bson import ObjectId
from pydantic import BaseModel, Field, model_validator


class ServiceCreate(BaseModel):
    name: str = Field(..., min_length=2, max_length=200)
    description: Optional[str] = None
    price: Optional[float] = None
    is_active: bool = True


class ServiceUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    price: Optional[float] = None
    is_active: Optional[bool] = None


class ServiceResponse(BaseModel):
    id: str = Field(alias="_id")
    name: str
    description: Optional[str] = None
    price: Optional[float] = None
    is_active: bool
    task_ids: list[str] = []
    created_at: datetime
    updated_at: datetime

    @model_validator(mode="before")
    @classmethod
    def convert_object_id(cls, data):
        if isinstance(data, dict):
            if "_id" in data and isinstance(data["_id"], ObjectId):
                data["_id"] = str(data["_id"])
            if "task_ids" in data and data["task_ids"]:
                data["task_ids"] = [
                    str(t) if isinstance(t, ObjectId) else t
                    for t in data["task_ids"]
                ]
        return data
